### PR TITLE
Add vietnam gov data 

### PR DESCRIPTION
**Pull request by phamleduy04**

### CODE REVIEW

File: scrapers/govScrapers/credentials/vietnam.pem [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+P0/HEZrIqPW1N+8QRcZs2eBelSaz662jue5/DJpmNXMyYE7l3YphLG5SEXdoltMY
+dVEVABt0iN3hxzgEQyjpFv3ZBdRdRydg1vs4O2xyopT4Qhrf7W8GjEXCBgCq5Ojc
+2bXhc3js9iPc0d1sjhqPpepUfJa3w/5Vjo1JXvxku88+vZbrac2/4EjxYoIQ5QxG
+V/Iz2tDIY+3GH5QFlkoakdH368+PUq4NCNk+qKBR6cGHdNXJ93SrLlP7u3r7l+L4
+HyaPs9Kg4DdbKDsx5Q5XLVq4rXmsXiBmGqW5prU5wfWYQ//u+aen/e7KJD2AFsQX
+j4rBYKEMrltDR5FL1ZoXX/nUh8HCjLfn4g8wGTeGrODcQgPmlKidrv0PJFGUzpII
+0fxQ8ANAe4hZ7Q7drNJ3gjTcBpUC2JD5Leo31Rpg0Gcg19hCC0Wvgmje3WYkN5Ap
+lBlGGSW4gNfL1IYoakRwJiNiqZ+Gb7+6kHDSVneFeO/qJakXzlByjAA6quPbYzSf
++AZxAeKCINT+b72x
+-----END CERTIFICATE-----
```
Comment (puf17640) [inline scrapers/govScrapers/credentials/vietnam.pem:124]: Why?

File: scrapers/govScrapers/getGovData.js [hunk 2]
```diff
@@ -6,8 +6,10 @@ const switzerlandData = require('./getSwitzerland');
 const nigeriaData = require('./getNigeria');
 const ukData = require('./getUK');
 const indiaData = require('./getIndia');
+const vietnamData = require('./getVietnam');
 const logger = require('../../utils/logger');
 
+
```
Comment (puf17640) [inline scrapers/govScrapers/getGovData.js:12]: too much empty lines

File: scrapers/govScrapers/getVietnam.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const axios = require('axios');
+const cherrio = require('cheerio');
+const rootCas = require('ssl-root-cas').create();
```
Comment (puf17640) [inline scrapers/govScrapers/getVietnam.js:3]: remove

File: scrapers/govScrapers/getVietnam.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const https = require('https');
+const { resolve } = require('path');
+const logger = require('../../utils/logger');
+const slugify = require('slugify');
+rootCas.addFile(resolve(__dirname, '..', 'govScrapers', 'credentials', 'vietnam.pem'));
+const httpsAgent = new https.Agent({ ca: rootCas });
+const columns = ['City', 'Cases', 'BeintgTreated', 'Recovered', 'Deaths'];
```
Comment (puf17640) [inline scrapers/govScrapers/getVietnam.js:10]: typo: BeintgTreated -> BeingTreated

Comment (puf17640) [reply scrapers/govScrapers/getVietnam.js:10]: also property names should be camelcased not titlecased

File: scrapers/govScrapers/getVietnam.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+const vietnamData = async () => {
+	try {
+		const html = cherrio.load((await axios.get('https://ncov.moh.gov.vn/', { httpsAgent })).data);
+		return html('table#sailorTable').children('tbody:first-of-type').children('tr').map(maprows).get().filter(el => !el.City.startsWith('BN'));
```
Comment (puf17640) [inline scrapers/govScrapers/getVietnam.js:14]: mapRows

File: scrapers/govScrapers/getVietnam.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	} catch (err) {
+		logger.err('Error: Requesting Vietnam Gov Data failed!', err);
+		return null;
+	}
+};
+
+const maprows = (_, row) => {
```
Comment (puf17640) [inline scrapers/govScrapers/getVietnam.js:21]: camelcase

File: scrapers/govScrapers/getVietnam.js (hunk 5 above)
Comment (puf17640) [reply scrapers/govScrapers/getVietnam.js:14]: also why do you filter `.filter(el => !el.City.startsWith('BN'))`

File: scrapers/govScrapers/getVietnam.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+	const city = { updated: Date.now() };
+	cherrio(row).children('td').each((index, cell) => {
+		cell = cherrio.load(cell);
+		switch (index) {
+			case 0: {
+				city[columns[index]] = slugify(cell.text(), { locale: 'vi' });
```
Comment (puf17640) [inline scrapers/govScrapers/getVietnam.js:27]: take slugify out

File: package.json [hunk 8]
```diff
@@ -43,6 +43,8 @@
     "express": "^4.16.3",
     "ioredis": "*",
     "mailgun-js": "^0.22.0",
+    "slugify": "^1.4.0",
+    "ssl-root-cas": "^1.3.1",
```
Comment (puf17640) [inline package.json:47]: not needed

File: tests/api_gov.spec.js [hunk 9]
```diff
@@ -256,6 +256,27 @@ describe('TESTING /v2/gov/austria', () => {
...
+            .end((err, res) => {
+                should.not.exist(err);
+                should.exist(res);
+                res.should.have.status(200);
+                res.body.should.be.a('array');
+                res.body.forEach((element) => {
+                    element.should.have.property('updated');
+                    element.should.have.property('city');
+                    element.should.have.property('cases');
+                    element.should.have.property('beingtreated');
```
Comment (puf17640) [inline tests/api_gov.spec.js:272]: this should be 'beingTreated' (also only works if you make the properties camelcase not titlecase..

**Comment by phamleduy04** [comment]
I tested and it's work perfectly fine.

**Comment by phamleduy04** [comment]
I mean when i get /v2/gov/vietnam it's return data

**Comment by puf17640** [comment]
I understand that it might work on your machine, but as you can see, it does not work outside your environment. The problem you are getting and why you added the SSL cert is that the vietnamese government apparently can't configure SSL correctly and their chain is incomplete. 

I pulled your changes and will try to find a fix for this.

**Comment by phamleduy04** [comment]
Ok thanks

**Comment by phamleduy04** [comment]
Can you try to create a certificate by this [link on stackoverflow](https://stackoverflow.com/questions/31673587/error-unable-to-verify-the-first-certificate-in-nodejs)

(You scroll to sch's answer)

**Comment by puf17640** [comment]
That's the point, I don't want us to have to create a certificate. Check my PR for details.

**Pull request closed without merging**